Lib/test/test_weakref.py

```diff
@@ -116,6 +116,20 @@ def test_basic_ref(self):
         del o
         repr(wr)
 
+    @support.cpython_only
+    def test_ref_repr(self):
+        obj = C()
+        ref = weakref.ref(obj)
+        self.assertEqual(repr(ref),
+                         f"<weakref at {id(ref):#x}; "
+                         f"to '{C.__module__}.{C.__qualname__}' "
+                         f"at {id(obj):#x}>")
+
+        obj = None
+        gc_collect()
+        self.assertEqual(repr(ref),
+                         f'<weakref at {id(ref):#x}; dead>')
+
```

Comment:
Is it worth to adding a test case for a class which defines the `__name__` attribute? 
(the repr would be slightly changed due of this)

Reply:
I added a test for a type with a ``__name__`` property.